File: python/gush_connect.py
---

### **2. `python/gush_connect.py`**

Create a folder named `python` and add `gush_connect.py`:

```python
"""
Gush Connect Python SDK v1.0
Official Python client library for Gush AI Agent Workspace (https://ai.sstore.ng/)
"""

import json
import logging
from typing import Any, Dict, Optional
import requests

logging.basicConfig(level=logging.INFO)

class GushConnectClient:
    def __init__(self, workspace_url: str = "https://ai.sstore.ng/", auth_token: Optional[str] = None):
        self.workspace_url = workspace_url.rstrip("/") + "/"
        self.auth_token = auth_token
        self.session = requests.Session()
        self.session.headers.update({
            "Content-Type": "application/json",
            "User-Agent": "GushConnect-PythonSDK/1.0"
        })
        if self.auth_token:
            self.session.headers.update({"Authorization": f"Bearer {self.auth_token}"})

    def trigger_agent_task(
        self,
        prompt: str,
        mode: str = "general",
        model: str = "gemini::gemini-2.5-flash",
        search_pref: str = "allow",
        project_id: int = 0
    ) -> Dict[str, Any]:
        """
        Dispatches a user or automated prompt to the Gush AI Agent engine.
        """
        endpoint = f"{self.workspace_url}?action="
        payload = {
            "messages": [{"role": "user", "content": prompt}],
            "mode": mode,
            "model": model,
            "search_pref": search_pref,
            "project_id": project_id
        }

        try:
            response = self.session.post(endpoint, json=payload, timeout=90)
            response.raise_for_status()
            return response.json()
        except requests.exceptions.RequestException as e:
            logging.error(f"Gush Connect Request Error: {e}")
            return {"ok": False, "error": str(e)}

    def execute_integration_tool(self, tool_id: int, integration_id: int, params: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
        """
        Executes a registered user API integration tool programmatically.
        """
        endpoint = f"{self.workspace_url}?action=test_user_integration_tool"
        payload = {
            "tool_id": tool_id,
            "integration_id": integration_id,
            "params": params or {}
        }

        try:
            response = self.session.post(endpoint, json=payload, timeout=45)
            response.raise_for_status()
            return response.json()
        except requests.exceptions.RequestException as e:
            logging.error(f"Tool Execution Error: {e}")
            return {"ok": False, "error": str(e)}


if __name__ == "__main__":
    # Starter Verification Execution
    client = GushConnectClient()
    res = client.trigger_agent_task("System Status Check via Gush Connect SDK.")
    print("Execution Result:", json.dumps(res, indent=2))

File: javascript/gush_connect.js
/**
 * Gush Connect JavaScript / Node.js SDK v1.0
 * Official JS client library for Gush AI Agent Workspace (https://ai.sstore.ng/)
 */

class GushConnectClient {
  constructor(options = {}) {
    this.workspaceUrl = (options.workspaceUrl || 'https://ai.sstore.ng/').replace(/\/+$/, '') + '/';
    this.authToken = options.authToken || null;
  }

  /**
   * Helper method to send JSON HTTP requests
   */
  async _request(action, payload = {}) {
    const url = `${this.workspaceUrl}?action=${action}`;
    const headers = {
      'Content-Type': 'application/json',
      'User-Agent': 'GushConnect-JSSDK/1.0'
    };

    if (this.authToken) {
      headers['Authorization'] = `Bearer ${this.authToken}`;
    }

    try {
      const response = await fetch(url, {
        method: 'POST',
        headers,
        body: JSON.stringify(payload)
      });

      if (!response.ok) {
        const errorText = await response.text();
        throw new Error(`HTTP Error ${response.status}: ${errorText}`);
      }

      return await response.json();
    } catch (err) {
      return { ok: false, error: err.message };
    }
  }

  /**
   * Dispatch prompt task to Gush AI Agent
   */
  async triggerAgentTask(prompt, config = {}) {
    const payload = {
      messages: [{ role: 'user', content: prompt }],
      mode: config.mode || 'general',
      model: config.model || 'gemini::gemini-2.5-flash',
      search_pref: config.searchPref || 'allow',
      project_id: config.projectId || 0
    };

    return await this._request('', payload);
  }

  /**
   * Execute registered user API tool
   */
  async executeTool(toolId, integrationId, params = {}) {
    return await this._request('test_user_integration_tool', {
      tool_id: toolId,
      integration_id: integrationId,
      params
    });
  }
}

// Export for ES modules and Node.js environments
if (typeof module !== 'undefined' && module.exports) {
  module.exports = { GushConnectClient };
}
